components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -304,6 +305,22 @@ fn cot(arg: &Option<Real>) -> Result<Option<Real>> {
     }
 }
 
+#[inline]
+#[rpn_fn]
+fn pow(lhs: &Option<Real>, rhs: &Option<Real>) -> Result<Option<Real>> {
+    match (lhs, rhs) {
+        (Some(lhs), Some(rhs)) => {
+            let pow = (lhs.into_inner()).pow(rhs.into_inner());
+            if pow.is_infinite() {
+                Err(Error::overflow("DOUBLE", format!("{}.pow({})", lhs, rhs)).into())
```

Comment:
I think we should handle overflow with an `EvalContext`

Reply:
Shoud I add an input param like `&mut EvalContext `?

Reply:
Hi, I have added `EvalContext` handle overflow, please take a look. Thanks

Reply:
Hi, I notice there are many vectorized methods  handle overflow without `EvalContext`, so it seems we don't have a common way about error handing.
And when I add `EvalContext` in `pow` function, the push down test failed and I don't know how to fix it:

 NoPushDown Output: 
Error 1690: DOUBLE value is out of range in 'pow(1983, 12795)'

PushDownWithoutVec Output: 
Error 1690: DOUBLE value is out of range in '1983.pow(12795)'

PushDownWithVec Output: 
field1	field2
0	NULL
0	NULL
0	NULL
0	NULL
0	NULL
0	NULL
0	NULL
0	NULL
0	NULL
0	NULL
0	NULL